haystack/components/agents/agent.py

```diff
@@ -84,6 +85,7 @@ def __init__(
         :param raise_on_tool_invocation_failure: Should the agent raise an exception when a tool invocation fails?
             If set to False, the exception will be turned into a chat message and passed to the LLM.
         :param streaming_callback: A callback that will be invoked when a response is streamed from the LLM.
+        :param stream_tool_result: If True, the tool result will be streamed to the streaming callback.
```

Comment:
I might expand on this a bit to explain what we mean by streaming here. Which is what you told me offline that the whole result will be sent to streaming_callback once it's available.